src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -84,8 +84,10 @@ private void writeAck(ChannelHandlerContext ctx, byte protocol, int sequence) {
     }
 
     private void clientTimeout() {
-        logger.debug("Client Timeout");
-        this.context.close();
+        if(!processing.get()) {
```

Comment:
We have test to cover that will check that a connection is close when the timeout is reached its in the `ServerTest`